### PR TITLE
Add DICOM Archive README/Spec

### PR DESCRIPTION
**Pull request by driusan**
Add specification for the DICOM archive in the standard format.

### CODE REVIEW

File: modules/dicom_archive/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Dicom Archive
+
+## Purpose
+
+The DICOM Archive module provides a front end to view the metadata
+about DICOMs which have been inserted into the LORIS pipeline.
```
Comment (cmadjar) [inline modules/dicom_archive/README.md]: Maybe rephrase "into the LORIS pipeline" as "into the LORIS database via the imaging pipeline"

File: modules/dicom_archive/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+The DICOM Archive shows DICOM tar metadata. It does not provide a
+way to visualize scans or view metadata of other data formats. It
+provides a frontend to the `tarchive_*` tables in LORIS, but does
+not directly inspect files on the filesystem.
+
+## Permissions
+
+The permission `dicom_archive_view_allsites` is required to access
+the DICOM Archive module.
```
Comment (cmadjar) [inline modules/dicom_archive/README.md]: There are no permissions for own site in this module? Note for later, could be useful, no?

File: modules/dicom_archive/README.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+Patient ID column.
+
+The `showTransferStatus` configuration option is obsolete and should
+not be used, but determines if a first "Transfer Status" column
+appears in the menu table.
+
+## Interactions with LORIS
+
+The tarchive tables used by the DICOM archive must be populated by
+the imaging `dicomTar` script before there is any data in the module.
```
Comment (cmadjar) [inline modules/dicom_archive/README.md]: add the extension at 'dicomTar.pl'

File: modules/dicom_archive/README.md (hunk 2 above)
Comment (driusan) [reply modules/dicom_archive/README.md]: yeah, but currently the hasAccess just checks for "dicom_archive_view_allsites"

File: modules/dicom_archive/README.md (hunk 1 above)
Comment (driusan) [reply modules/dicom_archive/README.md]: good catch, done.

File: modules/dicom_archive/README.md (hunk 3 above)
Comment (driusan) [reply modules/dicom_archive/README.md]: done

**Comment by driusan** [comment]
I just realized the correction isn't right. The failed images are still in the dicom archive, so it needs some reworking still..

**Comment by driusan** [comment]
@cmadjar I just reworded again, how does this version sound?

File: modules/dicom_archive/README.md [hunk 4]
```diff
@@ -3,8 +3,8 @@
 ## Purpose
 
 The DICOM Archive module provides a front end to view the metadata
-about DICOMs which have been inserted into the LORIS database via
-the imaging pipeline.
+about DICOMs which have been processed by the LORIS database via
```
Comment (cmadjar) [inline modules/dicom_archive/README.md]: @driusan I don't really like the "processed" term. The imaging pipeline does not "process" the images, it simply registers the DICOM information into the SQL database. No processing involved. Does that make sense?

Comment (driusan) [reply modules/dicom_archive/README.md]: I agree, but I don't like the word "inserted" either because it appears in the dicom archive even if it wasn't inserted into the imaging browser when it fails the pipeline, which is what people interpret as "inserted into LORIS".

Do you (or anyone) have any other suggestions?

Comment (cmadjar) [reply modules/dicom_archive/README.md]: What about something like:
"The DICOM Archive module provides a front end to view the metadata about DICOM studies and series that have been registered in the LORIS database via the first step of the imaging pipeline (using dicomTar.pl). The information displayed in this module includes all modalities of the DICOM studies (including both failed and successful acquisitions). Protocol checks and QC are performed later on and not represented in this module."

Comment (MounaSafiHarab) [reply modules/dicom_archive/README.md]: I would refrain from talkibg about failed and successful acquisitions as the DICOM archive precedes these steps. 
I would say something like:’ the module displays metadata of the Uploaded/transferred scans’ DICOMs the way they are archived in the various tarchive_* tables by the pipeline insertion scripts, I.e. at the study, series, and file levels’

Comment (driusan) [reply modules/dicom_archive/README.md]: What if I just left out the second last sentence in Cecile's proposal?

"The DICOM Archive module provides a front end to view the metadata about DICOM studies and series that have been registered in the LORIS database via the first step of the imaging pipeline (using dicomTar.pl).  Protocol checks and QC are performed later on and not represented in this module."

Comment (cmadjar) [reply modules/dicom_archive/README.md]: Agree with Mouna, the failed and successful acquisitions might be interpreted in different ways.

Your suggestion about removing the second last sentence sounds good to me!

**Comment by MounaSafiHarab** [comment]
2 last comments:

1) the only thing about the module that maybe we need to mention is the "protocol" column that is added and that is "computed" in the php scripts based on what the MRI scripts do in the subsequent steps.

2) perhaps mention something about the md5sum since the module displays it? you mention earlier that the module does not perform any file inspection, which is true, but an md5sum is in place to track changes to the .tar

File: modules/dicom_archive/README.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+The DICOM Archive is primarily used by DCC staff who are investigating
+the status of imaging pipeline scripts.
+
+## Scope
+
+The DICOM Archive shows DICOM tar metadata. It does not provide a
+way to visualize scans or view metadata of other data formats. It
+provides a frontend to the `tarchive_*` tables in LORIS, but does
+not directly inspect files on the filesystem.
```
Comment (MounaSafiHarab) [inline modules/dicom_archive/README.md]: perhaps mentioning something about the md5sum since the module displays it (in the same context that the module does not inspect the files on the filesystem but tracks changes to the .tar through tarchive_validation.pl later on).
JUst a thought but might be unnecessary if it complicates the text.

File: modules/dicom_archive/README.md (hunk 4 above)
Comment (MounaSafiHarab) [reply modules/dicom_archive/README.md]: sounds good to me too!!!

Comment (driusan) [reply modules/dicom_archive/README.md]: I put that in, can you guys reread it and review it as a whole?

File: modules/dicom_archive/README.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
+# Dicom Archive
+
+## Purpose
+
+The DICOM Archive module provides a front end to view the metadata
+about DICOM studies and series that have been registered in the
+LORIS database via the first step of the imaging pipeline (using
+`dicomTar.pl`). Protocol checks and QC are performed later on and not
+represented in this module."
```
Comment (cmadjar) [inline modules/dicom_archive/README.md]: Get rid of the " at the end of the sentence

File: modules/dicom_archive/README.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+The `patientIDRegex` configuration has a similar purpose for the
+Patient ID column.
+
+The `showTransferStatus` configuration option is obsolete and should
+not be used, but determines if a first "Transfer Status" column
+appears in the menu table.
+
+## Interactions with LORIS
+
```
[the remark this answers is not in the document]
Comment (cmadjar) [reply modules/dicom_archive/README.md]: Agreed.

File: modules/dicom_archive/README.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+The permission `dicom_archive_view_allsites` is required to access
+the DICOM Archive module.
+
+## Configurations
+
+The `patientNameRegex`, `LegoPhantomRegex`, and `LivingPhantomRegex`
+configuration variables provide regular expressions to ensure that
+the DICOM has been properly anonymized. If at least one of these
+isn't matched, the Patient Name is displayed as "INVALID - HIDDEN".
+
```
[the remark this answers is not in the document]
Comment (driusan) [reply modules/dicom_archive/README.md]: I don't know if that's the intended behaviour, or just because a blank regex matches everything.

File: modules/dicom_archive/README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (driusan) [reply modules/dicom_archive/README.md]: done

File: modules/dicom_archive/README.md (hunk 5 above)
Comment (driusan) [reply modules/dicom_archive/README.md]: I think that might be too specific and be distracting in this overview document.

**Comment by driusan** [comment]
@cmadjar @MounaSafiHarab I don't know why I can't reply to the last thread, but there's now a sentence about the link to the imaging browser too.

File: modules/dicom_archive/README.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (MounaSafiHarab) [reply modules/dicom_archive/README.md]: I only mention this because for projects that do not touch this field in the configuration module, some default behavior happens (and I have seen in some demos for projects that has PatientID information which displayed some identifying info because they only anonymized the PatientName field); so it might be a good place to let projects know that they have the option  to replace it by Hidden by doing the proper setting.

**Comment by MounaSafiHarab** [comment]
it is all good for me.

File: modules/dicom_archive/README.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (driusan) [reply modules/dicom_archive/README.md]: ok, good point.

[the remark this answers is not in the document]
Comment (driusan) [reply modules/dicom_archive/README.md]: I added something.

File: modules/dicom_archive/README.md (hunk 2 above)
Comment (mohadesz) [reply modules/dicom_archive/README.md]: Should we create a Redmine task for this?

Comment (driusan) [reply modules/dicom_archive/README.md]: Sure.
